### PR TITLE
HRCPP-208 Hotrod execute() implementation

### PR DESCRIPTION
**Pull request by rigazilla**
Implementation of the execute() operation and a test for it in Simple.cpp

### CODE REVIEW

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - WIN_S2012 [Build 282](http://ci.infinispan.org/viewLog.html?buildId=35032&buildTypeId=HotRodCpp_PullRequestsMonitorWinS2012&guest=1) for f1bdbcc366955dd49769498d64dc9d9d57e27d0d
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:00:04

Process exited with code 1

File: test/Simple.cpp [hunk 1]
```diff
@@ -251,17 +244,101 @@ int basicTest(int argc, char** argv, const char* protocolVersion, Marshaller<K>
...
+            "var cache = cacheManager.getCache();\n"
+            "cache.put(\"a\", \"abc\");\n"
+            "cache.put(\"b\", \"b\");\n"
+            "cache.get(\"a\");\n");
+            std::string script_name("script.js");
+            cacheManager.getCache<std::string,std::string>(km,
+                    &Marshaller<std::string>::destroy,
+                    vm,
+                    &Marshaller<std::string>::destroy,"___script_cache").put(script_name, script);
+            char* execResult = cache.execute(script_name,s);
```
Comment (tristantarrant) [inline test/Simple.cpp]: I guess the result is in JBossMarshalled format. It would be nice to unmarshall it instead of inspecting it as below.

**Comment by rigazilla** [comment]
The execute returns a sequence of bytes (as java client does), so I wrote the test in this general way. Maybe I can keep the test general and adding an helper that remove the jboss preamble. But the bytes should be compared in this way (or equivalent strncmp).
or
I can suppose the script to return simple string or int (as actually does but this is not the point) and unmarshall the data according, but it seems to me that this doesn't add value to the test.
wdyt

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 392](http://ci.infinispan.org/viewLog.html?buildId=35033&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for f1bdbcc366955dd49769498d64dc9d9d57e27d0d
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:08:59

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 393](http://ci.infinispan.org/viewLog.html?buildId=35034&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for f1bdbcc366955dd49769498d64dc9d9d57e27d0d
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:08:10

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - WIN_S2012 [Build 294](http://ci.infinispan.org/viewLog.html?buildId=35515&buildTypeId=HotRodCpp_PullRequestsMonitorWinS2012&guest=1) for f1bdbcc366955dd49769498d64dc9d9d57e27d0d
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:01:28

Process exited with code 1

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - WIN_S2012 [Build 296](http://ci.infinispan.org/viewLog.html?buildId=35524&buildTypeId=HotRodCpp_PullRequestsMonitorWinS2012&guest=1) for 8d011b4213c5cce3276ad99d0259466c32319c2c
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:01:55

Process exited with code 1

**Comment by tristantarrant** [comment]
@rigazilla you need to adapt this to the ScopedBuffer removal

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 403](http://ci.infinispan.org/viewLog.html?buildId=35518&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 6a86de81ba353592401a102cb4c58b8f5d42557b
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:08:21

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 404](http://ci.infinispan.org/viewLog.html?buildId=35519&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 6a86de81ba353592401a102cb4c58b8f5d42557b
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:00:10

Process exited with code 2

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 407](http://ci.infinispan.org/viewLog.html?buildId=35525&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 6a86de81ba353592401a102cb4c58b8f5d42557b
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:08:30

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 408](http://ci.infinispan.org/viewLog.html?buildId=35526&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 6a86de81ba353592401a102cb4c58b8f5d42557b
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:00:11

Process exited with code 2

**Comment by rigazilla** [comment]
Related branch is messed up... Created a new pull request

**Pull request closed without merging**